flink-java/src/main/java/org/apache/flink/api/java/typeutils/TypeExtractor.java

```diff
@@ -391,6 +392,10 @@ protected TypeExtractor() {
 				curT = typeToClass(curT).getGenericSuperclass();
 			}
 			
+			if(typeToClass(curT).equals(Tuple0.class)) {
```

Comment:
No, calling `typeToClass()` with a TypeVar will throw an `IllegalArgumentException`. `typeToClass()` should only be used if  `isClassType()` is true.